src/test/java/seedu/address/model/company/internship/UniqueInternshipListTest.java

```diff
@@ -0,0 +1,315 @@
+        Internship secondInternship = new InternshipBuilder(DATA_ANALYST_WITH_DATETIME).build();
+
+        uniqueInternshipList.add(secondInternship);
+        uniqueInternshipList.add(firstInternship);
+
+        List<Internship> replacementInternshipList = new ArrayList<>();
+        Internship thirdInternship = new InternshipBuilder(SOFTWARE_ENGINEER_WITHOUT_DATETIME).build();
+        Internship forthInternship = new InternshipBuilder(SOFTWARE_ENGINEER_WITHOUT_DATETIME).build();
+        replacementInternshipList.add(thirdInternship);
+        replacementInternshipList.add(forthInternship);
+
+        assertThrows(DuplicateInternshipException.class, () -> uniqueInternshipList
+                .setInternships(replacementInternshipList));
+    }
+
+    @Test
+    public void setInternshipsWithNullList_throwsNullPointerException() {
+        UniqueInternshipList uniqueInternshipList = new UniqueInternshipList();
+        Internship firstInternship = new InternshipBuilder(SOFTWARE_ENGINEER_WITH_DATETIME).build();
+        Internship secondInternship = new InternshipBuilder(DATA_ANALYST_WITH_DATETIME).build();
+
+        uniqueInternshipList.add(secondInternship);
+        uniqueInternshipList.add(firstInternship);
+
+
+        assertThrows(NullPointerException.class, ()->uniqueInternshipList.setInternships((List<Internship>) null));
+    }
+
+    @Test
+    public void iterator_validInternships_success() {
+        UniqueInternshipList uniqueInternshipList = new UniqueInternshipList();
+        List<Internship> typicalInternships = new ArrayList<>();
+        Internship firstInternship = new InternshipBuilder(SOFTWARE_ENGINEER_WITH_DATETIME).build();
+        Internship secondInternship = new InternshipBuilder(DATA_ANALYST_WITH_DATETIME).build();
+
+        typicalInternships.add(secondInternship);
+        typicalInternships.add(firstInternship);
+
+        uniqueInternshipList.setInternships(typicalInternships);
+
+        // Test that the iterator correctly iterates over the internships
+        Iterator<Internship> iterator = uniqueInternshipList.iterator();
+        for (Internship internship : typicalInternships) {
+            assertTrue(iterator.hasNext());
+            assertEquals(internship, iterator.next());
+        }
+    }
+
+    @Test
+    public void equals_sameObject_true() {
+        UniqueInternshipList uniqueInternshipList = new UniqueInternshipList();
+        uniqueInternshipList.add(new InternshipBuilder().build());
+
+        // Both references point to the same object, so they should be equal
+        assertTrue(uniqueInternshipList.equals(uniqueInternshipList));
+    }
+
+    @Test
+    public void equals_differentTypes_false() {
+        UniqueInternshipList uniqueInternshipList = new UniqueInternshipList();
+
+        // Comparing with a different type should return false
+        assertFalse(uniqueInternshipList.equals(5));
+    }
+
+    @Test
+    public void equals_null_false() {
```

Comment:
You might want to be more specific with the naming here. I feel like `nullInternshipList` might be good here